config/filesystems.php

```diff
@@ -59,11 +59,31 @@
             'driver' => 's3',
             'key' => env('AWS_ACCESS_KEY_ID'),
             'secret' => env('AWS_SECRET_ACCESS_KEY'),
-            'region' => env('AWS_DEFAULT_REGION'),
-            'bucket' => env('AWS_BUCKET'),
-            'url' => env('AWS_URL'),
+            'region' => env('AWS_DEFAULT_REGION', 'us-east-1'),
+            'bucket' => env('AWS_BUCKET', get_aws_bucket_from_env()),
+            'url' => env('AWS_URL', get_aws_url_from_env()),
         ],
 
     ],
 
 ];
+
+function get_url_path($url) {
+    $parsed = parse_url($url);
+    return array_key_exists('path', $parsed) ? $parsed['path'] : '';
+}
+
+function get_url_without_path($url, $path=null) {
+    $path = $path ?? '';
+    return substr($url, 0, strlen($url) - strlen($path));
+}
+
+function get_aws_bucket_from_env() {
```

Comment:
Can you provide a comment here with an example URL that's being parsed? I'm not familiar with the strings set by the heroku addon and it'd be helpful to have that here as a reference when I'm looking at this code in the future

Reply:
I gave examples in the commits, but I am now looking to extract this to a helper to use in the setup controller so we can write to config vars and not parse per-request 👍

Reply:
Also tests 👍